File: example-databinding/src/main/java/com/xwray/groupie/example/databinding/item/UpdatableItem.java
package com.xwray.groupie.example.databinding.item;

public class UpdatableItem extends SmallCardItem {

    private final int index;

    public UpdatableItem(int index) {
        super(String.valueOf(index));
        this.index = index;
    }

    @Override
    public long getId() {
        return index;
    }
}

File: example-viewbinding/src/main/java/com/xwray/groupie/example/viewbinding/item/UpdatableItem.kt
package com.xwray.groupie.example.viewbinding.item

import androidx.annotation.ColorInt

class UpdatableItem(
    @ColorInt colorInt: Int,
    private val index: Int
) : SmallCardItem(colorInt, index.toString()) {

    override fun getId(): Long = index.toLong()
}
